game/missiongenerator/luagenerator.py

```diff
@@ -307,3 +172,141 @@ def inject_plugins(self) -> None:
             if plugin.enabled:
                 plugin.inject_scripts(self)
                 plugin.inject_configuration(self)
+
+
+class LuaValue:
+    key: Optional[str]
+    value: List[str]
+    is_array = False
+
+    def __init__(self, key: Optional[str], values: List[str]):
+        self.key = key
+        self.value = values
+
+    def escape_value(self, value: str) -> str:
+        value = value.replace('"', "'")  # Replace Double Quote as this is the delimiter
+        value = value.replace(os.sep, "/")  # Replace Backslash as path separator
+        return '"{0}"'.format(value)
+
+    def serialize(self) -> str:
+        serialized_value = ""
+        if self.key:
+            serialized_value += self.key + " = "
+        if self.value:
+            if self.is_array:
+                escaped_values = [self.escape_value(v) for v in self.value]
+                serialized_value += "{" + ", ".join(escaped_values) + "}"
+            else:
+                serialized_value += self.escape_value(self.value[0])
+        else:
+            serialized_value += "{}"
+        return serialized_value
+
+
+class LuaItem(ABC):
+    values: List[LuaValue]
+    name: Optional[str] = None
+    is_single_value = False
+
+    def __init__(self, name: Optional[str]):
+        self.values = []
+        self.name = name
+
+    def set_value(self, value: str) -> None:
+        self.is_single_value = True
+        self.values = [LuaValue(None, [value])]
+
+    def add_data_array(self, key: str, values: List[str]) -> None:
+        value = LuaValue(key, values)
+        value.is_array = True
+        self.values.append(value)
+
+    def add_key_value(self, key: str, value: str) -> None:
+        self.values.append(LuaValue(key, [value]))
+
+    @abstractmethod
+    def add_item(self, item_name: Optional[str] = None) -> LuaItem:
+        pass
+
+    @abstractmethod
+    def add_object(self, object_name: Optional[str] = None) -> LuaItem:
+        pass
+
+    @abstractmethod
+    def serialize(self) -> str:
+        serialized_data = [d.serialize() for d in self.values]
+        if self.is_single_value:
+            return serialized_data[0]
+        if len(self.values) > 0:
+            return "{" + ", ".join(serialized_data) + "}"
+        return "{}"
+
+
+class LuaData(LuaItem):
+    objects: List[LuaData]
+    base_name: Optional[str] = None
+
+    def __init__(self, name: Optional[str], is_base_name: bool = True):
+        self.objects = []
+        if is_base_name:
+            self.base_name = name
+        super().__init__(name)
+
+    def add_item(self, item_name: Optional[str] = None) -> LuaItem:
+        item = LuaData(item_name, False)
+        self.objects.append(item)
+        return item
+
+    def add_object(self, object_name: Optional[str] = None) -> LuaItem:
```

Comment:
the name could be misleading as it returns an object which is a) newly created if it does not exist or b) reuses an existing one. But i am not really sure what would fit better as a function name